server.go

```diff
@@ -423,3 +433,33 @@ func WithRouteOptions(options ...func(*BaseRoute)) func(*Server) {
 		s.routeOptions = append(s.routeOptions, options...)
 	}
 }
+
```

Comment:
Curious about other thoughts on if all these were combined similar to how `WithOpenAPIConfig`

Reply:
Yeah I think this may work. I've implemented it so you guy can review if thats something you prefer.

Its cleaner than having a bunch of WithX